browser/brave_content_browser_client.cc

```diff
@@ -840,11 +828,6 @@ void BraveContentBrowserClient::RegisterBrowserInterfaceBindersForFrame(
   content::RegisterWebUIControllerInterfaceBinder<
       brave_new_tab_page::mojom::PageHandlerFactory, BraveNewTabUI>(map);
 #endif
-
-#if BUILDFLAG(ENABLE_PLAYLIST)
-  map->Add<playlist::mojom::PlaylistMediaHandler>(
-      base::BindRepeating(&BindPlaylistMediaHandler));
-#endif  // BUILDFLAG(ENABLE_PLAYLIST)
```

Comment:
Depends on what you want to achieve, I guess.

`RenderFrameHostReceiverSet<>` makes the cardinalities very clear:
```mermaid
erDiagram
    WebContents ||--|| PlaylistMediaHandler : ""
    PlaylistMediaHandler ||--o{ "mojom::PlaylistMediaResponder" : "via RenderFrameHostReceiverSet<>"
```
That is, for every `WebContents` there's a `PlaylistMediaHandler` (since it's a `WebContentsUserData<>`). And then each `PlaylistMediaHandler` takes care of processing media coming from remote `RenderFrame`s of the corresponding `WebContents` (as the receivers are routed to the right `PlaylistMediaHandler` when they are bound).

Although we don't need the messages on the `mojom::PlaylistMediaResponder` interface to retain FIFO with respect to legacy Chrome IPC messages, `RenderFrameHostReceiverSet<>` uses Channel-associated interface receivers regardless, and hence the registration is in `RegisterAssociatedInterfaceBindersForRenderFrameHost()`.

Alternatively, we could roll our own `RenderFrameHostReceiverSet<>` (with a `mojo::ReceiverSet<>`), and do the registration in `RegisterBrowserInterfaceBindersForFrame()`.